yaml_interfaces.md

```diff
@@ -0,0 +1,164 @@
+# GeoIPS Yaml-based Plugin Options
+Internally, we would like Plugins to behave the same, regardless of whether they are Module-based or Yaml-based. This
+means that we would like to automatically discover Plugins of both types and automatically create Python objects for
+each Plugin. This will allow them to share functionality without needing to duplicate code.
+
+## Current Situation
+
+### Module-based Plugins
+At present, we have updated GeoIPS to automatically discover and objectify Module-based plugins. This is achieved by
+implementing a class for each GeoIPS Interface and a second class for Plugins associated with each interface. Classes
+include:
+- `BaseInterface` and `BaseInterfacePlugin`: Base classes which should not be instantiated directly.
+- `AlgorithmsInterface` and `AlgorithmsInterfacePlugin`: Classes for the algorithms interface and algorithm plugins.
```

Comment:
@jsolbrig @cpcamacho AlgorithmsInterfacePlugin appears to be commented out currently?  Will that be implemented in the future?  Does it relate to an existing Issue?

Reply:
Sorry, any classes named `<Interface>Plugin` are currently cruft. This relates to #75. Apparently I needed to do some cleanup before making this PR... These are all currently generated automatically.

Reply:
Removed commented out <Interface>Plugin code.